susemanager-utils/susemanager-sls/src/modules/ssh_agent.py

```diff
@@ -0,0 +1,84 @@
+import logging
+from salt.exceptions import CommandExecutionError
+try:
+    from salt.utils.path import which_bin as _which_bin
+except ImportError:
+    from salt.utils import which_bin as _which_bin
+
+log = logging.getLogger(__name__)
+
+__virtualname__ = 'ssh_agent'
+
+__start_agent = '/usr/bin/ssh-agent'
+__ssh_add = '/usr/bin/ssh-add {}'
+
+def __virtual__():
+    '''
+    This module is always enabled while 'ssh-agent' is available.
+    '''
+    return __virtualname__ if _which_bin(['ssh-agent']) else (False, 'ssh-agent is not available')
+
+# def __call_ssh_tool(ssh_tool, cmd_args, **kwargs):
+#     log.debug("Calling ssh-agent: 'skuba {}' - Timeout: {}".format(cmd_args, timeout))
+#     try:
+#         ssh_tool_proc = salt.utils.timed_subprocess.TimedProc(
+#             [ssh_tool] + cmd_args.split(),
+#             stdout=subprocess.PIPE,
+#             stderr=subprocess.PIPE,
+#             timeout=timeout
+#         )
+#         ssh_tool_proc.run()
+#     except Exception as exc:
+#         error_msg = "Unexpected error while calling {}: {}".format(ssh_tool, exc)
+#         log.error(error_msg)
+#         raise CommandExecutionError(error_msg)    
+
+#     if ssh_tool_proc.process.returncode != 0 or ssh_tool_proc.stderr:
+#         error_msg = "Unexpected error {} when calling {}: {}".format(
+#                 ssh_tool_proc.returncode,
+#                 ssh_tool,
+#                 salt.utils.stringutils.to_str(ssh_tool_proc.stderr))
+#         log.error(error_msg)
+#         raise CommandExecutionError(error_msg)
+
+
+def start_agent(**kwargs):
+    # TODO add socket param
+    result = __salt__['cmd.run_all'](__start_agent, output_loglevel='quiet')
+    if result['retcode'] != 0:
```

Comment:
Using `cmd.run_all` is highly risky here. I would consider using the above commented function. I'll elaborate why in one of the following comments